### DISCUSSION TITLE
ESP_DMX on ESP32-S2

### DISCUSSION THREAD

**Discussion by fedechi** [someweisguy/esp_dmx#81] [General]
Hi Mitch,
first of all thanks for your work!
I'm trying, in very optimistic vay, to use the library with ESP32-S2 without success (v3.1).
Will the library be compatible with these devices in the future? 
or am I making some mistake.
thanks
Federico

**Comment by someweisguy** [comment]
Thanks for the kind words! Could you tell me more about the problem you are running into? This library should work with ESP32-S2. I am able to successfully build it for the S2 though I can't see if there are any runtime errors because I do not own an S2 myself.

Also, you should keep in mind that I haven't officially released v3.1 just yet. I am currently testing the current branch before officially releasing it. When I am ready to release a stable version, I will publish a tagged commit and the library will be available on the Arduino and PIO library managers.

**Comment by fedechi** [comment]
yeah great work. i'm working on custom board with ESP32-S2FNR4R2 for build responder, for test i use Enttec DMX USB PRO MK2.
I'm sure the hardware works from previous tests.
I use ESP-IDF 4.4.2 and ESP-PROG with jtag for debug and programm. The library version is 3.1.
simply try to execute the example idf_rdm_responder.

when programming the output console show
Info : esp32s2: Target halted, PC=0x400236D5, debug_reason=00000001

when i try to debug the PC fall immediatly in panic 
panic_abort(const char *details) with detail 97

if i go back on commit of lib (for example 15 jun 2023) 
the software run, i can debug and see DMX packet correctly but RDM not work (no discovery).
sorry, I don't currently have the ability to debug this type of device in depth.
thanks for support
F.

**Comment by someweisguy** [comment]
Thank you! I've opened issue #82 to track this problem!

I've ordered an ESP32-S2 to do some testing. I am, however, in the middle of moving across the country so it may be a while before I am able to solve this problem.

**Comment by someweisguy** [comment]
Just checking - do you get this error when you receive DMX data?

**Comment by fedechi** [comment]
yes, in the last release the error occurs immediately. the code not start.

**Comment by someweisguy** [comment]
Interesting. My ESP32-S2 arrived in the mail and I was able to flash it with the latest commit on the release/v3.1 branch. I've already packed my soldering iron, so I haven't tried testing to see if it responds to RDM, but it compiles and waits for RDM data without crashing. 

Maybe you can try the latest commit in the release/v3.1 branch and see if it works now? I have searched for the error code you're getting but I haven't found any useful information.
